src/crossroads/main.py

```diff
@@ -185,8 +191,80 @@ def validate_json(json_content):
     return True
 
 
-def data_which_overlap():
-    pass
+def data_which_overlap(first_timestamp_list, second_timestamp_list):
+    """
+    Checks that the dates in both dictionary lists have an overlap.
+
+    If the lists have an overlap, returns two new lists with only the same dates.
+    If the lists do not have an overlap but the closest dates are less than 7 days apart,
+    returns two new lists that contain only those dates.
+    If the lists do not have an overlap and the closest dates are more than 7 days apart,
+    returns 'None'.
+
+    :param first_timestamp_list: (list) first user dictionaries list with coordinates
+    :param second_timestamp_list: (list) second user dictionaries list with coordinates
+    :return: (lists or None) two dictionaries lists with overlapping coordinates, or 'None' if time ranges do not match
+    """
+    first_new_list = first_timestamp_list.copy()
+    second_new_list = second_timestamp_list.copy()
+
+    first_date_from_first = datetime.fromtimestamp(int(first_timestamp_list[0]['timestamp']) / 1000)
+    last_date_from_first = datetime.fromtimestamp(int(first_timestamp_list[-1]['timestamp']) / 1000)
+    first_date_from_second = datetime.fromtimestamp(int(second_timestamp_list[0]['timestamp']) / 1000)
+    last_date_from_second = datetime.fromtimestamp(int(second_timestamp_list[-1]['timestamp']) / 1000)
+
+    if last_date_from_first < first_date_from_second or last_date_from_second < first_date_from_first:
+        if int((first_date_from_first - last_date_from_second).days) in range(0, 8):
+            first_new_list = first_new_list[0:1]
+            second_new_list = second_new_list[-1:]
+            return first_new_list, second_new_list
+        elif int((first_date_from_second - last_date_from_first).days) in range(0, 8):
+            first_new_list = first_new_list[-1:]
+            second_new_list = second_new_list[0:1]
+            return first_new_list, second_new_list
+        else:
+            return None
```

Comment:
It's better to keep return values from function always the same amount and type. In that way it's more manageble, testable and readble. So instead `return None` you can use `return [None, None]`.